### PR TITLE
Add: eslint security plugin

### PR DESCRIPTION
**Pull request by MrSlimCoder**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

### CODE REVIEW

**Comment by viveksharmaui** [comment]
Hey! i added eslint security and found 342 problems (18 errors, 324 warnings) in the codebase i think we should fix it is related to security.

**Comment by viveksharmaui** [comment]
I want to fix all warnings and bugs from codebase so should I continue to work on this PR?

**Comment by viveksharmaui** [comment]
Guy's added a fix commit which fix `generic object injection sink` warning.

File: tools/translation/translate-challenges.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ function processFile(
           let transIndex = 0;
           testsArray.forEach((test, index) => {
             if (test.includes('- text')) {
-              testsArray[index] = '  - text: ' + translation[parseInt(transIndex)];
+              testsArray[parseIn(index)] = '  - text: ' + translation[parseInt(transIndex)];
```
Comment (Sky020) [inline tools/translation/translate-challenges.js]: Watch out for typos 😉

Comment (viveksharmaui) [reply tools/translation/translate-challenges.js]: This is outdated comment i have fixed this typo .

**Pull request closed without merging**